recipes/qtxlsxwriter/all/CMakeLists.txt

```
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(qtxlsxwriter)
+
+find_package(qt REQUIRED CONFIG)
+
+set(QTXLSXWRITER_TARGET_NAME qtxlsxwriter)
+
+include(conan_qt_executables_variables)
+set(CMAKE_AUTOMOC ON)
+
+file(GLOB QTXLSXWRITER_HEADERS LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.h)
+file(GLOB QTXLSXWRITER_SOURCES LIST_DIRECTORIES=false source_subfolder/src/xlsx/*.cpp)
+
+add_library(${QTXLSXWRITER_TARGET_NAME}
+	${QTXLSXWRITER_HEADERS}
+	${QTXLSXWRITER_SOURCES}
+)
+
+set(QT_VERSION "${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}.${QT_VERSION_PATCH}")
+target_include_directories(${QTXLSXWRITER_TARGET_NAME}
+	PRIVATE
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/src/xlsx>
+		# qtxlsxwriter uses qt private headers
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtCore/${QT_VERSION}>
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtGui/${QT_VERSION}>
+		$<BUILD_INTERFACE:${QT_ROOT}/include/QtGui/${QT_VERSION}/QtGui>
+	INTERFACE
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/source_subfolder/src/xlsx>
+		$<INSTALL_INTERFACE:include>
+)
+
+target_compile_features(${QTXLSXWRITER_TARGET_NAME}
+	PRIVATE
+		cxx_range_for
+)
+
+# place qt::Gui before qt::Core (important for some toolchains)
+# since the current qt recipe provides an incorrect order of libraries
+target_link_libraries(${QTXLSXWRITER_TARGET_NAME}
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_RELEASE>:CONAN_LIB::qt_Qt5Gui_RELEASE>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_DEBUG>:CONAN_LIB::qt_Qt5Gui_DEBUG>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_RELWITHDEBINFO>:CONAN_LIB::qt_Qt5Gui_RELWITHDEBINFO>
+	$<$<TARGET_EXISTS:CONAN_LIB::qt_Qt5Gui_MINSIZEREL>:CONAN_LIB::qt_Qt5Gui_MINSIZEREL>
+)
```

❓ Is this possible, it would seem odd Conan does not handle this already

```suggestion
target_link_libraries(${QTXLSXWRITER_TARGET_NAME} CONAN_LIB::qt_Qt5Gui)
```

---

This won't work because there is no target with the name `CONAN_LIB::qt_Qt5Gui`. In `conan_package_library_targets()` `CMakeDeps` creates several targets adding a build type to their names, and it seems these targets are not meant to be used. The current qt5 recipe doesn't support components and provides only the `qt::qt` target, but I also have to use these "intermediate" targets (like `CONAN_LIB::qt_Qt5Gui_RELEASE`) to deal with the problem relating to the order of libraries provided by the qt recipe.